### DISCUSSION TITLE
Using the output of tj-actions / changed-files in a loop for containers

### DISCUSSION THREAD

**Discussion by ServerGeek82** [tj-actions/changed-files#1149] [Q&A] [answered]
I noticed a few issues that were opened using the output of ${{ steps.changed-files-specific.outputs.all_changed_files }} in a loop.

Would it be possible to use the results in a remote shell script for docker compose?

cd /home/docker
docker compose -f ${{ steps.changed-files-specific.outputs.all_changed_files }} pull 

docker compose -f ${{ steps.changed-files-specific.outputs.all_changed_files }} up -d

**Comment by ServerGeek82** [comment]
Was thinking something like this might work, but would like to have a loop to run through the loop of changed YAML files. 

`

- name: Get changed files in the docs folder
  id: changed-files-specific
     uses: tj-actions/changed-files@v35
    with:
      files: compose-files/*.{yml,yaml}   # Alternatively using `docs/**` or `docs`
    
  - name: copy files from git to mxprod01.io
    uses: https://github.com/appleboy/scp-action@v0.1.4
    with:
      host: ${{ secrets.MXPROD01 }}
      username: ${{ secrets.USERNAME }}
      key: ${{ secrets.DEPLOY_KEY }}
      port: 22
      source: ${{ steps.changed-files.outputs.all_changed_files }}
      target: /home/docker/compose-files/

- name:  Testing update of container
--
  | uses: https://github.com/appleboy/ssh-action@v0.1.10
  | with:
  | host: ${{ secrets.MXPROD01 }}
  | username: ${{ secrets.USERNAME }}
  | password: ${{ secrets.PASSWORD }}
  | port: 22
  | script: |
  | hostname
  | cd /home/docker/compose-files
  | docker compose -f ${{ steps.changed-composed-files.outputs.all_changed_files }} up -d

`

**Comment by jackton1** [comment, marked as answer]
@ServerGeek82 The output of `all_changed_files` likely includes multiple files separated by spaces.

It can be passed to docker-compose with multiple -f options. e.g `-f [FILE1] -f [FILE2]`

Example:

```yml
      - name: Get docker-compose options
        run: |
          ALL_CHANGED_FILES=(${{ steps.changed-files-specific.outputs.all_changed_files }})
          ARGS=""
          for file in "${ALL_CHANGED_FILES[@]}"; do
            ARGS+="-f $file"
          done
         # Pass the args to docker-compose
```

> **Warning**:
> 
> The id value should match what's used in the changed-files action in this case it should be `changed-files-specific`.
